src/MultiMessageSender.sol

```diff
@@ -78,69 +86,81 @@ contract MultiMessageSender {
+
+        if (adapterLength == 0) {
+            revert Error.NO_SENDER_ADAPTER_FOUND();
+        }
 
-        uint256 adaptersChainId = 0;
+        /// @dev increments nonce
+        ++nonce;
 
-        /// default adapters
-        if (senderAdapters[_dstChainId].length > 0) {
-            /// if different set of adapters are configured for this desitnation chain
-            adaptersChainId = _dstChainId;
-        }
-        address[] storage adapters = senderAdapters[adaptersChainId];
-        /// send copies of the message through multiple bridges
-        for (uint256 i; i < adapters.length; ++i) {
-            message.bridgeName = IBridgeSenderAdapter(adapters[i]).name();
-            data = abi.encodeWithSelector(IMultiMessageReceiver.receiveMessage.selector, message);
-            uint256 fee =
-                IBridgeSenderAdapter(adapters[i]).getMessageFee(uint256(_dstChainId), _multiMessageReceiver, data);
-            /// if one bridge is paused it shouldn't halt the process
+        MessageLibrary.Message memory message =
+            MessageLibrary.Message(_dstChainId, _target, nonce, _callData, _expiration, "");
+
+        for (uint256 i; i < adapterLength;) {
+            IBridgeSenderAdapter bridgeAdapter = IBridgeSenderAdapter(adapters[i]);
+            message.bridgeName = bridgeAdapter.name();
+
+            /// @dev assumes CREATE2 deployment for mma sender & receiver
+            uint256 fee = bridgeAdapter.getMessageFee(_dstChainId, gac.getMultiMessageReceiver(), abi.encode(message));
+
+            /// @dev if one bridge is paused, the flow shouldn't be broken
             try IBridgeSenderAdapter(adapters[i]).dispatchMessage{value: fee}(
-                uint256(_dstChainId), _multiMessageReceiver, data
-            ) {
-                totalFee += fee;
-            } catch {
+                _dstChainId, gac.getMultiMessageReceiver(), abi.encode(message)
+            ) {} catch {
                 emit ErrorSendMessage(adapters[i], message);
             }
+
+            unchecked {
+                ++i;
+            }
         }
-        bytes32 msgId = MessageStruct.computeMsgId(message, uint64(block.chainid));
-        emit MultiMessageMsgSent(msgId, nonce, _dstChainId, _target, _callData, _expiration, adapters);
-        nonce++;
-        /// refund remaining native token to msg.sender
-        if (totalFee < msg.value) {
-            _safeTransferETH(msg.sender, msg.value - totalFee);
+
+        bytes32 msgId = MessageLibrary.computeMsgId(message, block.chainid);
+
+        /// refund remaining fee
+        /// FIXME: add an explicit refund address config
+        if (address(this).balance > 0) {
+            _safeTransferETH(msg.sender, address(this).balance);
         }
+
+        emit MultiMessageMsgSent(msgId, nonce, _dstChainId, _target, _callData, _expiration, adapters);
     }
 
     /// @notice Add bridge sender adapters
     /// @param _chainId is the destination chainId. Use 0 to add default adapers
```

Comment:
nit: reference to default adapters in the comment appears to be outdated now since defaults are not supported.

Reply:
if so, I suggest we check that `_chainId` is not 0